allennlp/models/model.py

```diff
@@ -333,6 +335,61 @@ def extend_embedder_vocab(self, extended_vocab: Vocabulary) -> None:
             if isinstance(module, (Embedding, TextFieldEmbedder)):
                 module.extend_vocab(extended_vocab)
 
+    def inspection_dict(self,
+                        inspect: str = "parameters",
+                        quite: bool = False) -> OrderedDict:
+        """
+        Returns ordered nested dict showing the module / parameters paths and (optionally)
+        corresponding module names of the model. This can be helpful to setup model path
+        based regex, for example in initializer. If quite is false, dict will be printed.
+        Both outputs are less verbose than print(model).
+
+        Parameters
+        ----------
+        inspect: str, optional, (default = "parameters")
+            Inspects "parameters" or "modules" of the model.
+        quite: bool, optional, (default = False)
+            If true, doesn't print the output, only returns the final dict.
+        """
+        if inspect not in ["parameters", "modules"]:
+            ConfigurationError("model inspection works with modules or parameters only")
+
+        items = self.named_modules() if inspect == "modules" else self.named_parameters()
+        module_dict = {model_path: module for model_path, module in items}
+        sorted_module_paths = sorted(module_dict.keys(), key=lambda path: path.count('.'))
+
+        def nested_set(dict_, keys, value):
```

Comment:
it's not clear to me what this function does (the docstring is not helpful, and its operation is pretty opaque)

also, why is this a nested function?